components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AccountConfirmationValidationHandler.java

```diff
@@ -117,4 +123,24 @@ private boolean isUserAccountConfirmed(User user) throws IdentityEventException
         return userConfirmed ;
     }
 
+    private boolean isUserExistsInDomain(UserStoreManager userStoreManager, String userName)
+            throws IdentityEventException {
+
+        boolean isExists = false;
+        try {
+            if (userStoreManager.isExistingUser(userName)) {
+                isExists = true;
+            }
+        } catch (UserStoreException e) {
+            throw new IdentityEventException("Error occurred while check user existence: " + userName, e);
```

Comment:
It doesn't mean much to throw an exception from  a private method. Better to handle userstore exception at calling level

Reply:
In fact there is a catch clause for userstore exception in calling level. So just throw as it is